### PR TITLE
[18.0][MIG] sale_product_set_packaging_qty: Migration to 18.0

### PR DESCRIPTION
**Pull request by trisdoan**
### Note
- depends on:
    - `https://github.com/OCA/product-attribute/pull/1758`
    - `https://github.com/OCA/sale-workflow/pull/3386`
 
### Changes in 18.0
- product_packaging was removed since 15.0, in [here](https://github.com/odoo/odoo/commit/688d65530575a6b2a69db404e3f11aaf84ef0035)
- constraint positive_qty was added since 16.0, in [here](https://github.com/odoo/odoo/commit/688d65530575a6b2a69db404e3f11aaf84ef0035)
   - As consequences, 
      - a check "Please select a packaging with a quantity bigger than 0" can be removed
      - test_error_write_qty_but_packaging_zero is not needed any longer

### CODE REVIEW

File: sale_product_set_packaging_qty/models/product_set.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    )
+    def _compute_product_packaging_qty(self):
+        for line in self:
+            uom_rounding = line.product_id.uom_id.rounding
+            if (
+                not line.product_packaging_id
+                or float_is_zero(line.quantity, precision_rounding=uom_rounding)
+                or float_is_zero(
+                    line.product_packaging_id.qty, precision_rounding=uom_rounding
+                )
```
Comment (xaviedoanhduy) [inline sale_product_set_packaging_qty/models/product_set.py]:
```suggestion
```
As mentioned in PR description, checking `qty` = 0 is not necessary

File: sale_product_set_packaging_qty/models/product_set.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                line.product_packaging_qty = 0
+                continue
+            line.product_packaging_qty = line.quantity / line.product_packaging_id.qty
+            line.update(line._prepare_product_packaging_qty_values())
+
+    def _inverse_product_packaging_qty(self):
+        for line in self:
+            if line.product_packaging_qty and not line.product_packaging_id:
+                raise UserError(
+                    _(
```
Comment (xaviedoanhduy) [inline sale_product_set_packaging_qty/models/product_set.py]:
```suggestion
                    self.env._(
```

File: sale_product_set_packaging_qty/models/product_set.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            )
+        else:
+            self.update({"product_packaging_qty": 0})
+
+    def prepare_sale_order_line_values(self, order, quantity, max_sequence=0):
+        res = super().prepare_sale_order_line_values(
+            order, quantity, max_sequence=max_sequence
+        )
+        if self.product_packaging_id:
+            res["product_packaging_id"] = self.product_packaging_id.id
```
Comment (xaviedoanhduy) [inline sale_product_set_packaging_qty/models/product_set.py:74]: `product_packaging_id` field on `sale.order.line` is a compute field. so do you think we should override that compute function or simply assign a new value to it.

https://github.com/odoo/odoo/blob/d1f0c66af06b4ca15cd91a44a03a9b13cdd8da16/addons/sale/models/sale_order_line.py#L209

**Review by xaviedoanhduy — CHANGES_REQUESTED** [review]

File: sale_product_set_packaging_qty/models/product_set.py (hunk 3 above)
Comment (trisdoan) [reply sale_product_set_packaging_qty/models/product_set.py:74]: It is a writable compute field, isn't it? Any advantage of overriding it?

**Review by xaviedoanhduy — APPROVED** [review]

**Pull request closed without merging**